catalog/app/containers/Auth/SSOOkta.js

```diff
@@ -0,0 +1,138 @@
+import { push } from 'connected-react-router/esm/immutable'
+import invariant from 'invariant'
+import * as React from 'react'
+import { FormattedMessage as FM } from 'react-intl'
+import * as redux from 'react-redux'
+import * as M from '@material-ui/core'
+
+import { useIntl } from 'containers/LanguageProvider'
+import * as Notifications from 'containers/Notifications'
+import * as Config from 'utils/Config'
+import * as NamedRoutes from 'utils/NamedRoutes'
+import * as Sentry from 'utils/Sentry'
+import defer from 'utils/defer'
+
+import * as actions from './actions'
+import * as errors from './errors'
+import msg from './messages'
+
+import oktaLogo from './okta-logo.svg'
+
+const MUTEX_POPUP = 'sso:okta:popup'
+const MUTEX_REQUEST = 'sso:okta:request'
+
+export default function SSOOkta({ mutex, next }) {
+  const cfg = Config.useConfig()
+  invariant(!!cfg.oktaClientId, 'Auth.SSO.Okta: config missing "oktaClientId"')
+  invariant(!!cfg.oktaCompanyName, 'Auth.SSO.Okta: config missing "oktaCompanyName"')
+
+  const sentry = Sentry.use()
+  const dispatch = redux.useDispatch()
+  const intl = useIntl()
+  const { push: notify } = Notifications.use()
+  const { urls } = NamedRoutes.use()
+
+  const handleClick = React.useCallback(() => {
+    if (mutex.current) return
+    mutex.claim(MUTEX_POPUP)
+
+    const oktaDomain = `https://${cfg.oktaCompanyName}.okta.com`
+    const nonce = '1' // TODO
```

Comment:
Let's do `hash(email + epochtime)`

Reply:
Oops... forgot to `git push`. Generating a random number now.